src/pymatgen/transformations/advanced_transformations.py

```diff
@@ -1488,64 +1498,112 @@ def apply_transformation(self, structure: Structure) -> Structure:
-            length5_vec = b - _proj(b, c)  # b-c plane
-            length6_vec = c - _proj(c, b)
-            length_vecs = np.array(
-                [
-                    length1_vec,
-                    length2_vec,
-                    length3_vec,
-                    length4_vec,
-                    length5_vec,
-                    length6_vec,
-                ]
-            )
+        if not self.allow_orthorhombic:
+            # boolean for if a sufficiently large supercell has been created
+            sc_not_found = True
+
+            target_sc_size = self.min_length
+            while sc_not_found:
+                target_sc_lat_vecs = np.eye(3, 3) * target_sc_size
+                length_vecs, n_atoms, superstructure, transformation_matrix = self.get_possible_supercell(
+                    lat_vecs, structure, target_sc_lat_vecs
+                )
 
-            # Get number of atoms
-            st = SupercellTransformation(self.transformation_matrix)
-            superstructure = st.apply_transformation(structure)
-            n_atoms = len(superstructure)
+                # Check if constraints are satisfied
+                if (
+                    np.min(np.linalg.norm(length_vecs, axis=1)) >= self.min_length
+                    and self.min_atoms <= n_atoms <= self.max_atoms
+                ) and (
+                    not self.force_90_degrees
+                    or np.all(np.absolute(np.array(superstructure.lattice.angles) - 90) < self.angle_tolerance)
+                ):
+                    self.transformation_matrix = transformation_matrix
+                    return superstructure
+
+                # Increase threshold until proposed supercell meets requirements
+                target_sc_size += self.step_size
+                self.check_exceptions(length_vecs, n_atoms)
+            raise AttributeError("Unable to find cubic supercell")
+
+        combined_list = [
+            [size_a, size_b, size_c]
+            for size_a in np.arange(self.min_length, self.max_length, self.step_size)
+            for size_b in np.arange(self.min_length, self.max_length, self.step_size)
+            for size_c in np.arange(self.min_length, self.max_length, self.step_size)
+        ]
+        combined_list = sorted(combined_list, key=sum)
 
+        for size_a, size_b, size_c in combined_list:
+            target_sc_lat_vecs = np.array([[size_a, 0, 0], [0, size_b, 0], [0, 0, size_c]])
+            length_vecs, n_atoms, superstructure, transformation_matrix = self.get_possible_supercell(
+                lat_vecs, structure, target_sc_lat_vecs
+            )
+            print(superstructure.lattice)
             # Check if constraints are satisfied
+            print(n_atoms)
             if (
                 np.min(np.linalg.norm(length_vecs, axis=1)) >= self.min_length
                 and self.min_atoms <= n_atoms <= self.max_atoms
             ) and (
                 not self.force_90_degrees
```

Comment:
@JaGeo , if I set `force_90_degrees` to True, I never leave the while loop

Reply:
That's not true. There is an additional statement here.

Reply:
` or np.all(np.absolute(np.array(superstructure.lattice.angles) - 90) < self.angle_tolerance)`

Reply:
the unit test runs forever without stopping when I set force_90_degrees to True for the orthorhombic supercell generation. the test finalized for the cubic supercell transformation.

Reply:
but not because of this. Can you test a bit more with min/max lengths?

Reply:
Oh sorry, my bad, I see now that this is even outside the while loop.

Reply:
No, this is really the stopping criterion but it should still stop or run into the attribute error if there is no cell like this.

Reply:
we can resolve this conversation, as step_size was the crucial part here

Reply:
Maybe add a hint to the doc that stepsize needs to be larger for the orthorhombic cells

Reply:
ok, done :)